### PR TITLE
Bug/1268 scala serialization on Hadoop

### LINKED ISSUE #1268: Scala deserialization exception for a job ran on Hadoop
I have ran the following job with Scatter plotter which is written in Scala: 

``` 
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<job xmlns="http://eobjects.org/analyzerbeans/job/1.0">
    <job-metadata>
        <job-description>Created with DataCleaner Enterprise edition 5.0-SNAPSHOT</job-description>
        <author>claudiap</author>
        <updated-date>2016-03-31+02:00</updated-date>
        <metadata-properties>
            <property name="CoordinatesX.datacleaner.customers.csv">40</property>
            <property name="CoordinatesY.datacleaner.customers.csv">40</property>
        </metadata-properties>
    </job-metadata>
    <source>
        <data-context ref="customers.csv_2"/>
        <columns>
            <column id="col_id" path="id" type="STRING"/>
            <column id="col_given_name" path="given_name" type="STRING"/>
            <column id="col_family_name" path="family_name" type="STRING"/>
            <column id="col_company" path="company" type="STRING"/>
            <column id="col_address_line" path="address_line" type="STRING"/>
            <column id="col_post_code" path="post_code" type="STRING"/>
            <column id="col_city" path="city" type="STRING"/>
            <column id="col_country" path="country" type="STRING"/>
            <column id="col_email" path="email" type="STRING"/>
            <column id="col_birthdate" path="birthdate" type="STRING"/>
            <column id="col_gender" path="gender" type="STRING"/>
            <column id="col_job_title" path="job_title" type="STRING"/>
            <column id="col_income_amount" path="income_amount" type="STRING"/>
            <column id="col_income_currency" path="income_currency" type="STRING"/>
        </columns>
    </source>
    <transformation>
        <transformer name="Birthdate as date">
            <descriptor ref="Convert to date"/>
            <metadata-properties>
                <property name="CoordinatesX">360</property>
                <property name="CoordinatesY">120</property>
            </metadata-properties>
            <properties>
                <property name="Time zone" value="Europe/Berlin"/>
                <property name="Null replacement" value="&lt;null&gt;"/>
                <property name="Date masks" value="[yyyy-MM-dd HH:mm:ss.S,yyyy-MM-dd HH:mm:ss,yyyy-MM-dd HH:mm,yyyyMMddHHmmssZ,yyMMddHHmmssZ,yyyy-MM-dd,yyyy.MM.dd,yyyy/MM/dd,dd-MM-yyyy,dd.MM.yyyy,dd/MM/yyyy,MM-dd-yyyy,MM.dd.yyyy,MM/dd/yyyy]"/>
            </properties>
            <input ref="col_birthdate"/>
            <output id="col_birthdateasdate" name="birthdate (as date)"/>
        </transformer>
        <transformer>
            <descriptor ref="Date to age"/>
            <metadata-properties>
                <property name="CoordinatesX">520</property>
                <property name="CoordinatesY">120</property>
            </metadata-properties>
            <properties/>
            <input ref="col_birthdateasdate"/>
            <output id="col_ageindays" name="Age in days"/>
            <output id="col_ageinyears" name="Age in years"/>
        </transformer>
        <transformer name="Money amount as number">
            <descriptor ref="Convert to number"/>
            <metadata-properties>
                <property name="CoordinatesX">520</property>
                <property name="CoordinatesY">200</property>
            </metadata-properties>
            <properties>
                <property name="Decimal separator" value="&amp;#44;"/>
                <property name="Minus sign" value="-"/>
                <property name="Null replacement" value="&lt;null&gt;"/>
                <property name="Thousand separator" value="."/>
            </properties>
            <input ref="col_income_amount"/>
            <output id="col_income_amountasnumber" name="income_amount (as number)"/>
        </transformer>
    </transformation>
    <analysis>
        <analyzer name="Age in years vs income_amount (as number)">
            <descriptor ref="Scatter plot"/>
            <metadata-properties>
                <property name="CoordinatesX">680</property>
                <property name="CoordinatesY">120</property>
            </metadata-properties>
            <properties/>
            <input ref="col_gender" name="Group column"/>
            <input ref="col_ageinyears" name="Variable1"/>
            <input ref="col_income_amountasnumber" name="Variable2"/>
        </analyzer>
    </analysis>
</job>

 ``` 

The **exception** while trying to open the  result file: 

```
Unexpected error!
 
Message:
    java.io.InvalidClassException: scala.collection.immutable.$colon$colon; local class incompatible: stream classdesc serialVersionUID = -8476791151983527571, local class serialVersionUID = 509929039250432923
Level:
    SEVERE
Stack Trace:
scala.collection.immutable.$colon$colon; local class incompatible: stream classdesc serialVersionUID = -8476791151983527571, local class serialVersionUID = 509929039250432923
    java.io.ObjectStreamClass.initNonProxy(Unknown Source)
    java.io.ObjectInputStream.readNonProxyDesc(Unknown Source)
    java.io.ObjectInputStream.readClassDesc(Unknown Source)
    java.io.ObjectInputStream.readOrdinaryObject(Unknown Source)
    java.io.ObjectInputStream.readObject0(Unknown Source)
    java.io.ObjectInputStream.defaultReadFields(Unknown Source)
    java.io.ObjectInputStream.readSerialData(Unknown Source)
    java.io.ObjectInputStream.readOrdinaryObject(Unknown Source)
    java.io.ObjectInputStream.readObject0(Unknown Source)
    java.io.ObjectInputStream.readObject(Unknown Source)
    java.util.HashMap.readObject(Unknown Source)
    sun.reflect.GeneratedMethodAccessor45.invoke(Unknown Source)
    sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    java.lang.reflect.Method.invoke(Unknown Source)
    java.io.ObjectStreamClass.invokeReadObject(Unknown Source)
    java.io.ObjectInputStream.readSerialData(Unknown Source)
    java.io.ObjectInputStream.readOrdinaryObject(Unknown Source)
    java.io.ObjectInputStream.readObject0(Unknown Source)
    java.io.ObjectInputStream.defaultReadFields(Unknown Source)
    java.io.ObjectInputStream.readSerialData(Unknown Source)
    java.io.ObjectInputStream.readOrdinaryObject(Unknown Source)
    java.io.ObjectInputStream.readObject0(Unknown Source)
    java.io.ObjectInputStream.readObject(Unknown Source)
    org.datacleaner.actions.OpenAnalysisJobActionListener.openAnalysisResult(OpenAnalysisJobActionListener.java:144)
    
```
The problem seems to be that Hadoop is using other version of Scala.

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #1268 

-  Transferred the analyzers and their results code from Scala to Java. 
-  Wizards work on Hadoop. 
-  <b>The old "analysis.result.dat" files will not be opened. </b>

### CODE REVIEW

**Comment by LosD** [comment]
Damn. I didn't consider that... I'm afraid we will have to support the old analysis.result.dat's as long as possible...

Can we give the Java-based Results a new class name, and use them by all components, but keep the old ones, marked with deprecated? That should make the transition painless for users (until the Scala version doesn't match anymore, of course).

**Comment by LosD** [comment]
For the `@Named` attribute, I'd prepend "Old " to it.

**Comment by ClaudiaPHI** [comment]
I will see what I can do. Only for results.

**Comment by LosD** [comment]
Honestly, I'm not a fan of changing the analyzers and renderers, unless there was a really good reason, it was just a waste of time, and increasing of risk.

**Comment by LosD** [comment]
Errr, I've no idea what I was on about with the `@Named` column, those are not for the results :)

File: components/visualization/src/main/scala/org/datacleaner/visualization/DensityAnalyzerChartScriptHeadElement.scala [hunk 1]
```diff
@@ -20,22 +25,26 @@ class DensityAnalyzerChartScriptHeadElement(result: DensityAnalyzerResult, eleme
   override def toHtml(context: HtmlRenderingContext): String = {
 
     val annotations = result.getRowAnnotations
```
Comment (LosD) [inline components/visualization/src/main/scala/org/datacleaner/visualization/DensityAnalyzerChartScriptHeadElement.scala]: If you `import collection.JavaConversions._` and define this like val annotations: `Map[(Int, Int) = result.getRowAnnotations`, you should be able to leave most of the code as it was.

http://www.scala-lang.org/docu/files/collections-api/collections_46.html

**Comment by LosD** [comment]
In general, I'm a bit afraid of all the changes in the tests (the new versions should be functionally equivalent as far as possible), but of course, if there's no way around them, so be it.

**Pull request closed without merging**